Tests/FunctionalTests/AEPEdgeResponseHandlerFunctionalTests.swift

```diff
@@ -0,0 +1,192 @@
+        Edge.sendEvent(experienceEvent: ExperienceEvent(xdm: ["eventType": "personalizationEvent", "test": "xdm"],
+                                                        data: nil),
+                       responseHandler: responseHandler1)
+        assertNetworkRequestsCount()
+        responseHandler1.await()
+
+        resetTestExpectations()
+        setExpectationNetworkRequest(url: FunctionalTestConst.EX_EDGE_INTERACT_URL_STR, httpMethod: HttpMethod.post, expectedCount: 1)
+        setNetworkResponseFor(url: FunctionalTestConst.EX_EDGE_INTERACT_URL_STR, httpMethod: HttpMethod.post, responseHttpConnection: httpConnection2)
+        Edge.sendEvent(experienceEvent: ExperienceEvent(xdm: ["eventType": "personalizationEvent", "test": "xdm"],
+                                                        data: nil),
+                       responseHandler: responseHandler2)
+
+        // verify
+        assertNetworkRequestsCount()
+        responseHandler2.await()
+
+        XCTAssertEqual(1, responseHandler1.onResponseHandles.count)
+        XCTAssertEqual(0, responseHandler1.onErrorHandles.count)
+        XCTAssertTrue(responseHandler1.onCompleteCalled)
+        XCTAssertEqual(0, responseHandler2.onResponseHandles.count)
+        XCTAssertEqual(2, responseHandler2.onErrorHandles.count)
+        XCTAssertTrue(responseHandler2.onCompleteCalled)
+    }
+
+    func testSendEvent_withEdgeResponseHandler_whenServerErrorAndHandle_callsOnError_callsOnResponse() {
+        let responseBodyWithHandleAndError = "\u{0000}{\"requestId\": \"0ee43289-4a4e-469a-bf5c-1d8186919a26\",\"handle\": [{\"payload\": [{\"id\": \"AT:eyJhY3Rpdml0eUlkIjoiMTE3NTg4IiwiZXhwZXJpZW5jZUlkIjoiMSJ9\",\"scope\": \"buttonColor\",\"items\": [{                           \"schema\": \"https://ns.adobe.com/personalization/json-content-item\",\"data\": {\"content\": {\"value\": \"#D41DBA\"}}}]}],\"type\": \"personalization:decisions\"}],\"errors\": [{\"message\": \"An error occurred while calling the 'X' service for this request. Please try again.\", \"code\": \"502\"}, {\"message\": \"An error occurred while calling the 'Y', service unavailable\", \"code\": \"503\"}]}\n"
+        let httpConnection: HttpConnection = HttpConnection(data: responseBodyWithHandleAndError.data(using: .utf8),
+                                                            response: HTTPURLResponse(url: edgeUrl,
+                                                                                      statusCode: 200,
+                                                                                      httpVersion: nil,
+                                                                                      headerFields: nil),
+                                                            error: nil)
+        let responseHandler = MockResponseHandler(expectedResponses: 1, expectedErrors: 1)
```

Comment:
Here you are expecting 2 error handles, however you initialize the MockResponseHandler with `expectedErrors: 1`.  Should the expected errors be 2?